### ISSUE TITLE
Can't install php5-mapscript or php-mapscript on php5.6

### ISSUE THREAD

**Issue by advcha** [oerdnj/deb.sury.org#461]
Hello, I use php 5.6.26-1+deb.sury.org~trusty+1. I use ppa:ondrej/php but I can't install either php5-mapscript or php-mapscript.
When I tried to install php5-mapscript, I got an error 'php5-mapscript : Depends: phpapi-20121212'
When I tried to install php-mapscript, I got an error 'the package is not available'.
How to make it work?

**Comment by oerdnj** [comment]
There's no package for `php-mapscript` as it's a part of bigger package (`mapserver`). You need to compile it manually by installing (preferably on clean chroot/dev machine):
```
sudo apt-get install php5.6-dev devscripts
sudo apt-get build-dep mapserver
sudo apt-get remove php5-common
apt-get source mapserver
cd mapserver-6.4.1/
```

Apply this patch:
```
--- debian/rules~	2014-02-28 15:54:20.000000000 +0000
+++ debian/rules	2016-09-26 14:19:52.177217057 +0000
@@ -22,7 +22,7 @@ CFLAGS+=$(CPPFLAGS)
 CFLAGS+=$(LDFLAGS)
 
 RUBYVERSIONS=$(shell dh_ruby --print-supported)
-PHP5API=$(shell php-config5 --phpapi)
+PHP5API=$(shell php-config5.6 --phpapi)
 MANPAGES:=$(wildcard debian/man/*.*.xml)
 
 MS_VERSION=$(shell dpkg-parsechangelog | sed -ne 's/^Version: \(.*\)-.*/\1/p')
--- debian/control~	2014-02-28 15:54:20.000000000 +0000
+++ debian/control	2016-09-26 14:20:28.385182799 +0000
@@ -21,7 +21,7 @@ Build-Depends: debhelper (>= 9),
                libgif-dev,
                librsvg2-dev,
                libpq-dev,
-               php5-dev,
+               php5.6-dev,
                swig,
                libperl-dev,
                python-all (>= 2.6.6-3~),
```

And run `debuild -uc -us`.

I can't test it myself right now as launchpad is not working at this exact moment, but hopefully this will give you enough material to try to do it yourself.

**Comment by advcha** [comment]
Thanks for the fast response oerdnj, I've run the above commands, but I stuck on the patch. How to apply the patch? what are the steps? sorry for my noob.

**Comment by oerdnj** [comment]
`patch -p1`
Paste the patch followed by Ctrl-D 

Or just make the changes in editor. 

Please use Askubuntu.com or Ubuntu forums for newbie questions. There's a chance you get somebody with more time (and patience) than me.

**Comment by oerdnj** [comment]
Correction, try with `patch -p0`

**Comment by advcha** [comment]
Thanks oerdnj, I also posted the same thing on askubuntu [(http://askubuntu.com/questions/829926/how-to-apply-patch-on-mapserver)](http://askubuntu.com/questions/829926/how-to-apply-patch-on-mapserver), it seems with the parameter -p0, mapserver can be patched 
`teddy@teddy-K43SJ:~/mapserver-6.4.1$ patch -p0 < oerdnj.patch`
`patching file debian/rules`
`patching file debian/control`

But then when I try to build with `debuild -uc -us`, it failed. Here is what I got on my linux terminal
`teddy@teddy-K43SJ:~/mapserver-6.4.1$ debuild -uc -us`
` dpkg-buildpackage -rfakeroot -D -us -uc`
`dpkg-buildpackage: source package mapserver`
`dpkg-buildpackage: source version 6.4.1-2`
`dpkg-buildpackage: source distribution unstable`
`dpkg-buildpackage: source changed by Bas Couwenberg <sebastic@xs4all.nl>`
 `dpkg-source --before-build mapserver-6.4.1`
`dpkg-buildpackage: host architecture amd64`
`   fakeroot debian/rules clean`
`dh clean --with python2,pkgkde_symbolshelper \`
`	         --parallel \`
`	         --buildsystem cmake`
`     dh_testdir -O--parallel -O--buildsystem=cmake`
`     debian/rules override_dh_auto_clean`
`make[1]: Entering directory '/home/teddy/mapserver-6.4.1'`
`dh_clean`
`rm -f -f /home/teddy/mapserver-6.4.1/debian/man/*.1`
`rm -f -rf obj-*/`
`rm -f -rf build-ruby*/ debian/tmp-ruby*/`
`make[1]: Leaving directory '/home/teddy/mapserver-6.4.1'`
`     dh_clean -O--parallel -O--buildsystem=cmake`
` dpkg-source -b mapserver-6.4.1`
`dpkg-source: info: using source format '3.0 (quilt)'`
`dpkg-source: info: building mapserver using existing ./mapserver_6.4.1.orig.tar.gz`
`dpkg-source: info: local changes detected, the modified files are:`
`  mapserver-6.4.1/oerdnj.patch`
`dpkg-source: error: aborting due to unexpected upstream changes, see /tmp/mapserver_6.4.1-2.diff.549Dd5`
`dpkg-source: info: you can integrate the local changes with dpkg-source --commit`
`dpkg-buildpackage: error: dpkg-source -b mapserver-6.4.1 gave error exit status 2`
`debuild: fatal error at line 1364:`
`dpkg-buildpackage -rfakeroot -D -us -uc failed`

**Comment by oerdnj** [comment]
You should start reading the logs... Just remove the `oerdnj.patch` from the mapserver directory...

**Comment by advcha** [comment]
Thanks. Sorry for bothering you again. Yes. that's work after i removed the patch file. but I got another error when I build the source at 82%. if you think the below log is not formatted correctly, pls see also the error in [pastebin](http://pastebin.com/m4pTj65B)
`teddy@teddy-K43SJ:~/mapserver-6.4.1$ debuild -uc -us `
`...`
```
[ 82%] Building C object mapscript/php/CMakeFiles/php_mapscript.dir/error.c.o
cd /home/teddy/mapserver-6.4.1/obj-x86_64-linux-gnu/mapscript/php && /usr/bin/x86_64-linux-gnu-gcc  -DCOMPILE_DL=1 -Dphp_mapscript_EXPORTS -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-z,now -D_FORTIFY_SOURCE=2  -Wall -Wdeclaration-after-statement -fPIC -I/home/teddy/mapserver-6.4.1/obj-x86_64-linux-gnu -I/home/teddy/mapserver-6.4.1/renderers/agg/include -I/usr/include/freetype2 -I/usr/include/fribidi -I/usr/include/cairo -I/usr/include/postgresql -I/usr/include/gdal -I/usr/include/librsvg-2.0 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng12 -I/usr/include/pixman-1 -I/usr/include/libxml2 -I/usr/include/php/20131226 -I/usr/include/php/20131226/main -I/usr/include/php/20131226/TSRM -I/usr/include/php/20131226/Zend -I/usr/include/php/20131226/ext -I/usr/include/php/20131226/ext/date/lib    -o CMakeFiles/php_mapscript.dir/error.c.o   -c /home/teddy/mapserver-6.4.1/mapscript/php/error.c
In file included from /usr/include/php/20131226/main/php.h:40:0,
                 from /home/teddy/mapserver-6.4.1/mapscript/php/php_mapscript.h:35,
                 from /home/teddy/mapserver-6.4.1/mapscript/php/error.c:32:
/home/teddy/mapserver-6.4.1/mapscript/php/error.c: In function ‘zim_errorObj___get’:
/usr/include/php/20131226/Zend/zend_API.h:582:14: error: called object ‘zend_error’ is not a function or function pointer
    zend_error(E_ERROR, "String size overflow"); \
              ^
/usr/include/php/20131226/Zend/zend_API.h:626:39: note: in expansion of macro ‘ZVAL_STRING’
 #define RETVAL_STRING(s, duplicate)   ZVAL_STRING(return_value, s, duplicate)
                                       ^
/home/teddy/mapserver-6.4.1/mapscript/php/php_mapscript_util.h:186:9: note: in expansion of macro ‘RETVAL_STRING’
         RETVAL_STRING( (value ? value:"") , 1);    \
         ^
/home/teddy/mapserver-6.4.1/mapscript/php/error.c:71:8: note: in expansion of macro ‘IF_GET_STRING’
   else IF_GET_STRING("routine", php_error->error->routine)
        ^
In file included from /home/teddy/mapserver-6.4.1/mapscript/php/php_mapscript.h:35:0,
                 from /home/teddy/mapserver-6.4.1/mapscript/php/error.c:32:
/usr/include/php/20131226/main/php.h:293:19: note: declared here
 #define php_error zend_error
                   ^
/home/teddy/mapserver-6.4.1/mapscript/php/error.c:58:21: note: in expansion of macro ‘php_error’
   php_error_object *php_error;
                     ^
In file included from /usr/include/php/20131226/main/php.h:40:0,
                 from /home/teddy/mapserver-6.4.1/mapscript/php/php_mapscript.h:35,
                 from /home/teddy/mapserver-6.4.1/mapscript/php/error.c:32:
/usr/include/php/20131226/Zend/zend_API.h:582:14: error: called object ‘zend_error’ is not a function or function pointer
    zend_error(E_ERROR, "String size overflow"); \
              ^
/usr/include/php/20131226/Zend/zend_API.h:626:39: note: in expansion of macro ‘ZVAL_STRING’
 #define RETVAL_STRING(s, duplicate)   ZVAL_STRING(return_value, s, duplicate)
                                       ^
/home/teddy/mapserver-6.4.1/mapscript/php/php_mapscript_util.h:186:9: note: in expansion of macro ‘RETVAL_STRING’
         RETVAL_STRING( (value ? value:"") , 1);    \
         ^
/home/teddy/mapserver-6.4.1/mapscript/php/error.c:72:10: note: in expansion of macro ‘IF_GET_STRING’
     else IF_GET_STRING("message", php_error->error->message)
          ^
In file included from /home/teddy/mapserver-6.4.1/mapscript/php/php_mapscript.h:35:0,
                 from /home/teddy/mapserver-6.4.1/mapscript/php/error.c:32:
/usr/include/php/20131226/main/php.h:293:19: note: declared here
 #define php_error zend_error
                   ^
/home/teddy/mapserver-6.4.1/mapscript/php/error.c:58:21: note: in expansion of macro ‘php_error’
   php_error_object *php_error;
                     ^
/home/teddy/mapserver-6.4.1/mapscript/php/error.c: In function ‘zim_errorObj___set’:
/usr/include/php/20131226/main/php.h:293:19: warning: variable ‘zend_error’ set but not used [-Wunused-but-set-variable]
 #define php_error zend_error
                   ^
/home/teddy/mapserver-6.4.1/mapscript/php/error.c:85:21: note: in expansion of macro ‘php_error’
   php_error_object *php_error;
                     ^
make[4]: *** [mapscript/php/CMakeFiles/php_mapscript.dir/error.c.o] Error 1
make[4]: Leaving directory `/home/teddy/mapserver-6.4.1/obj-x86_64-linux-gnu'
make[3]: *** [mapscript/php/CMakeFiles/php_mapscript.dir/all] Error 2
make[3]: Leaving directory `/home/teddy/mapserver-6.4.1/obj-x86_64-linux-gnu'
make[2]: *** [all] Error 2
make[2]: Leaving directory `/home/teddy/mapserver-6.4.1/obj-x86_64-linux-gnu'
dh_auto_build: make -j1 returned exit code 2
make[1]: *** [override_dh_auto_build] Error 2
make[1]: Leaving directory `/home/teddy/mapserver-6.4.1'
make: *** [build] Error 2
dpkg-buildpackage: error: debian/rules build gave error exit status 2
debuild: fatal error at line 1364:
dpkg-buildpackage -rfakeroot -D -us -uc failed
```
